go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -623,13 +623,37 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 		}()
 	}
 
-	// Now we know no writes are in-flight and no new writes can occur.
-	// We just need to wait for no write being blocked on semi-sync ACKs.
-	err = tm.SemiSyncMonitor.WaitUntilSemiSyncUnblocked(ctx)
+	isSemiSyncBlocked, err := tm.MysqlDaemon.IsSemiSyncBlocked(ctx)
```

Comment:
Why did we make this change? I'd really like to see us use the new and improved `tm.SemiSyncMonitor.WaitUntilSemiSyncUnblocked(ctx)` if we can. Are there reasons we can't?